src/dbwrapper.h

```diff
@@ -25,190 +23,145 @@ class dbwrapper_error : public std::runtime_error
     dbwrapper_error(const std::string& msg) : std::runtime_error(msg) {}
 };
 
-/** These should be considered an implementation detail of the specific database.
- */
-namespace dbwrapper_private {
-
-/** Handle database error by throwing dbwrapper_error exception.
- */
-void HandleError(const leveldb::Status& status);
+// this exception should be caught and handled in CDBWrapper
+class dbwrapper_notfound : public std::runtime_error {
+public:
+    dbwrapper_notfound() : std::runtime_error("db: item not found") { }
+};
 
+class ClearStreamRAII {
+public:
+    ClearStreamRAII(CDataStream& s1) : s1(&s1), s2(nullptr) { }
+    ClearStreamRAII(CDataStream& s1, CDataStream& s2) : s1(&s1), s2(&s2) { }
+    ~ClearStreamRAII() {
+        s1->clear();
+        if (s2 != nullptr) {
+            s2->clear();
```

Comment:
Hmm, wouldn't it be straightforward to use two `ClearStreamRAII` instead, one for each stream, and then simplify the implementation of this class?

(I guess I should (and will) stick with BU but @sickpig pointed me to this interesting PR - and this bit just strikes me as odd :D)

Reply:
That makes sense. No need to stick to BU 😁